app/models/manageiq/providers/amazon/storage_manager/ebs/refresh_parser.rb

```diff
@@ -77,6 +79,57 @@ def link_storage_associations
     end if @data[:cloud_volumes]
   end
 
+  def link_volume_to_disk(volume_hash, attachments)
+    uid = volume_hash[:ems_ref]
+
+    attachments.each do |a|
+      if a['device'].blank?
+        _log.warn "#{log_header}: Volume: #{uid}, is missing a mountpoint, skipping the volume processing"
+        _log.warn "#{log_header}:   EMS: #{@ems.name}, Instance: #{a['instance_id']}"
+        next
+      end
+
+      dev = File.basename(a['device'])
+
+      vm = @ems.parent_manager.vms.detect { |v| v.ems_ref == a['instance_id'] }
```

Comment:
@ems.parent_manager.vms.find_by(:ems_ref => a['instance_id'])

this will do 1 small query to the db, otherwise you are doing fetch me all vms and then O(n) search in Ruby. :-)